### PR TITLE
Orion 15.0.0-plat-9689-thumbnail vid sec and slice

### PR DESCRIPTION
**Pull request by ZurKaltura**
It also includes cache changes to the thumbnail service

### CODE REVIEW

File: plugins/thumbnail/lib/model/thumbStorage/kThumbStorageS3.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	protected function login()
+	{
+		$this->s3Mgr->login(self::$configParams[self::CONF_URL],
+							self::$configParams[self::CONF_USER_NAME],
+							self::$configParams[self::CONF_PASSWORD]);
+	}
+
+	public function saveFile($fileName, $content)
+	{
+		$this->s3Mgr->registerStreamWrapper();
```
Comment (MosheMaorKaltura) [inline plugins/thumbnail/lib/model/thumbStorage/kThumbStorageS3.php]: Move to consructor

File: plugins/thumbnail/lib/model/thumbStorage/kThumbStorageS3.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		$path = $this->getFullPath($fileName);
+		$this->url = 's3://' . $path;
```
Comment (MosheMaorKaltura) [inline plugins/thumbnail/lib/model/thumbStorage/kThumbStorageS3.php]: Move to getUrl()

File: plugins/thumbnail/services/ThumbnailService.php [hunk 3]
```diff
@@ -8,44 +8,50 @@
...
+		if(!$storage->loadFile($transformString, $lastModified))
+		{
+			$imagick = $transformation->execute();
+			$storage->saveFile($transformString, $imagick, $lastModified);
+		}
+
+		$storage->render($lastModified);
 	}
 
 	protected function saveTransformationResult($imagick)
```
Comment (MosheMaorKaltura) [inline plugins/thumbnail/services/ThumbnailService.php]: remove

File: plugins/thumbnail/services/ThumbnailService.php [hunk 4]
```diff
@@ -8,44 +8,50 @@
...
 		file_put_contents($tempFilePath, $imagick);
 		return $tempFilePath;
 	}
 
-	protected function parseUrl()
+	/**
+	 * @param $transformString
+	 * @return kImageTransformation
+	 */
+	protected function parseTransformString($transformString)
```
Comment (MosheMaorKaltura) [inline plugins/thumbnail/services/ThumbnailService.php]: Move to new class kThumbnailUrlParser

File: plugins/thumbnail/lib/model/imagickAction/kCompositeAction.php [hunk 5]
```diff
@@ -37,18 +41,20 @@ protected function validateInput()
 	{
 		if($this->opacity && ($this->opacity < self::MIN_OPACITY || $this->opacity > SELF::MAX_OPACITY))
 		{
-			throw new KalturaAPIException(KalturaThumbnailErrors::BAD_QUERY, 'opacity must be between 1-100');
+			$data = array("errorString" => 'opacity must be between 1-100');
```
[the remark this answers is not in the document]
Comment (ZurKaltura) [reply plugins/thumbnail/lib/model/imagickAction/kCompositeAction.php]: Will be done in another pull request

**Pull request merged**